site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,627 @@
+.. _HttpStatusException: apidocs/index.html?com/linecorp/armeria/server/HttpStatusException.html
+.. _JacksonRequestConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.html
+.. _PathMapping: apidocs/index.html?com/linecorp/armeria/server/PathMapping.html
+.. _RequestConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/RequestConverterFunction.html
+.. _ResponseConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/ResponseConverterFunction.html
+.. _ServerBuilder: apidocs/index.html?com/linecorp/armeria/server/ServerBuilder.html
+.. _Service: apidocs/index.html?com/linecorp/armeria/server/Service.html
+.. _StringRequestConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/StringRequestConverterFunction.html
+
+.. _server-annotated-service:
+
+Annotated HTTP Service
+======================
+
+*Armeria* provides a way to write an HTTP service using annotations. It helps a user to make his or her codes
+simple and easy to understand. A user is able to run an HTTP service by couple of lines codes using
+annotations as follows. ``getHello()`` method in the example would be mapped to the path of ``/hello``
+with an HTTP GET method.
+
+.. code-block:: java
+
+    ServerBuilder sb = new ServerBuilder();
+    sb.annotatedService(new Object() {
+        @Get("/hello/{name}")
+        public HttpResponse hello(@Param("name") String name) {
+            return HttpResponse.of(HttpStatus.OK,
+                                   MediaType.PLAIN_TEXT_UTF_8,
+                                   "Hello, %s!", name);
+        }
+    });
+
+Mapping HTTP service methods
+----------------------------
+
+To map each method in an annotated HTTP service class to HTTP path, both `@Path`_ annotation and HTTP method
+annotations are used. Here is the list of HTTP method annotations that *Armeria* supports. Each of them is
+mapped to the HTTP method which has the same name.
+
+- `@Get`_
+- `@Head`_
+- `@Post`_
+- `@Put`_
+- `@Delete`_
+- `@Options`_
+- `@Patch`_
+- `@Trace`_
+
+If you want to handle several types of HTTP methods in a single service method, you can annotate your
+service method as follows.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+        @Get
+        @Post
+        @Put
+        @Delete
+        @Path("/hello")
+        public HttpResponse hello() {
+            ...
+        }
+    }
+
+To handle a single HTTP method in a single service method, you can annotate your service method simply,
+as follows. In this case, you can omit `@Path`_ annotation and write your path with `@Get`_ annotation.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+        @Get("/hello")
+        public HttpResponse hello() {
+            ...
+        }
+    }
+
+There are 5 types of PathMapping_ s provided for describing a path with `@Path`_ annotation.
+
+- Exact mapping
```

Comment:
How about moving the examples after the mapping name, so that the examples are more visible?